Sources/ZIPFoundation/Archive.swift

```diff
@@ -111,8 +108,22 @@ public final class Archive: Sequence {
     public let accessMode: AccessMode
     var archiveFile: UnsafeMutablePointer<FILE>
     var endOfCentralDirectoryRecord: EndOfCentralDirectoryRecord
+    var zip64EndOfCentralDirectory: Zip64EndOfCentralDirectory?
     var preferredEncoding: String.Encoding?
 
+    var totalNumberOfEntriesInCentralDirectory: UInt {
+        zip64EndOfCentralDirectory?.record.totalNumberOfEntriesInCentralDirectory
+            ?? UInt(endOfCentralDirectoryRecord.totalNumberOfEntriesInCentralDirectory)
+    }
+    var sizeOfCentralDirectory: Int {
+        zip64EndOfCentralDirectory?.record.sizeOfCentralDirectory
+            ?? Int(endOfCentralDirectoryRecord.sizeOfCentralDirectory)
+    }
+    var offsetToStartOfCentralDirectory: Int {
+        zip64EndOfCentralDirectory?.record.offsetToStartOfCentralDirectory
+            ?? Int(endOfCentralDirectoryRecord.offsetToStartOfCentralDirectory)
+    }
```

Comment:
Good idea to provide ZIP/ZIP64 agnostic accessors here!
The type/size problem applies here as well, though. On 32-bit platforms, the `Int` will use the default word size there and counteract the ZIP64 benefits.